### PR TITLE
feat: [CDS-77851]: integrate save and run pipeline

### PR DESCRIPTION
**Pull request by PratyushGar**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/70-pipeline/components/PipelineStudio/CreateModal/PipelineCreate.tsx [hunk 1]
```diff
@@ -107,7 +107,7 @@ export default function CreatePipelines({
   const { isGitSyncEnabled, gitSyncEnabledOnlyForFF, supportingGitSimplification } = useAppStore()
   const oldGitSyncEnabled = isGitSyncEnabled && !gitSyncEnabledOnlyForFF
   const { trackEvent } = useTelemetry()
-  const [yamlVersion, setYamlVersion] = React.useState<YamlVersion>(yamlVersionValue)
+  const [yamlVersion, setYamlVersion] = React.useState<YamlVersion>(YamlVersion[1])
```
Comment (rameshx) [inline src/modules/70-pipeline/components/PipelineStudio/CreateModal/PipelineCreate.tsx:110]: Is this change intentional?

File: src/modules/70-pipeline/pages/pipeline-list/PipelineListUtils.ts [hunk 2]
```diff
@@ -46,7 +50,8 @@ export const getRouteProps = (pathParams: PipelineListPagePathParams, pipeline?:
     repoIdentifier: pipeline?.gitDetails?.repoIdentifier,
     repoName: pipeline?.gitDetails?.repoName,
     connectorRef: pipeline?.connectorRef,
-    storeType: pipeline?.storeType as StoreType
+    storeType: pipeline?.storeType as StoreType,
+    ...(pipeline?.yamlVersion === YamlVersion[1] && { yamlVersion: pipeline.yamlVersion })
```
Comment (rameshx) [inline src/modules/70-pipeline/pages/pipeline-list/PipelineListUtils.ts:54]: Did something similar in this PR https://github.com/harness/harness-core-ui/pull/17616

**Review by rameshx — APPROVED** [review]

**Review by srdjan-harness — APPROVED** [review]

**Review by srdjan-harness — APPROVED** [review]

File: src/modules/70-pipeline/components/PipelineStudio/CreateModal/PipelineCreate.tsx (hunk 1 above)
Comment (PratyushGar) [reply src/modules/70-pipeline/components/PipelineStudio/CreateModal/PipelineCreate.tsx:110]: Yes, we need to default it to V1, the default value in context is V0. As this modal only appears for the V1 pipeline and is behind FF, we can modify it later with common architecture.

File: src/modules/70-pipeline/pages/pipeline-list/PipelineListUtils.ts (hunk 2 above)
Comment (PratyushGar) [reply src/modules/70-pipeline/pages/pipeline-list/PipelineListUtils.ts:54]: 👍 I will rebase my PR again once that is merged and revert this change
